src/shaka_scheme/system/vm/HeapVirtualMachine.cpp

```diff
@@ -0,0 +1,178 @@
+//
+// Created by Billy Wooton on 10/2/17.
+//
+
+#include "shaka_scheme/system/vm/HeapVirtualMachine.hpp"
+#include "shaka_scheme/system/base/Environment.hpp"
+#include "shaka_scheme/system/vm/CallFrame.hpp"
+
+namespace shaka {
+
+HeapVirtualMachine::~HeapVirtualMachine() {}
+
+NodePtr HeapVirtualMachine::evaluate_assembly_instruction() {
```

Comment:
Please change this to return `void`. Since the only case where it returns an accumulator is when the instruction is `(halt)`, then that means the leaky `nullptr` returns will simple just invite people to segfault when they dereference it. Better to restrict this and let them query it through the getter.